src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -35,12 +39,28 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             return CommandResult.Executed;
         }
 
-        private void OnBrokerChanged(object sender, EventArgs e) {
+        private void OnRSessionInitialized(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
```

Comment:
I don't have a strong opinion on this. If there's some code that you suspect could be destabilized, let's keep it as is for now. But I think it's worthwhile to revisit the overall design at some later point, and see if we can simplify it to reduce the likelihood of future bugs around current semantics.